pkg/apis/core/v1alpha1/tiltrun_types.go

```diff
@@ -0,0 +1,274 @@
+	//
+	// If nil, the resource does not perform builds (for example, a local resource without a serve_cmd).
+	Build *BuildState `json:"build,omitempty"`
+	// Runtime provides information about the current execution of the resource.
+	Runtime RuntimeState `json:"runtime,omitempty"`
+}
+
+// BuildState includes details about a currently pending build, currently active build, and (last) terminated build.
+type BuildState struct {
+	// Pending gives details about the currently enqueued build (if any).
+	Pending *PendingBuild `json:"pending,omitempty"`
+	// Active gives details about the currently running build (if any).
+	Active *ActiveBuild `json:"active,omitempty"`
+	// Terminated gives details about the last finished build (if any).
+	Terminated *TerminatedBuild `json:"terminated,omitempty"`
+}
+
+// PendingBuild is a build that has been enqueued for execution but has not yet started.
+type PendingBuild struct {
+	// TriggerTime is when the earliest event occurred (e.g. file change) occurred that resulted in a build being
+	// enqueued.
+	TriggerTime metav1.MicroTime `json:"triggerTime"`
+	// Reason is a description for why the build is being triggered. There may be more than one cause, but only a
+	// single reason is provided.
+	Reason string `json:"reason"`
+}
+
+// ActiveBuild is a build that is currently running but has not yet finished.
+type ActiveBuild struct {
+	// StartTime is when the build began.
+	StartTime metav1.MicroTime `json:"startTime"`
+}
+
+// TerminatedBuild is a build that finished running, either because it completed successfully or encountered an error.
+type TerminatedBuild struct {
+	// StartTime is when the build began.
+	StartTime metav1.MicroTime `json:"startTime"`
+	// FinishTime is when the build stopped.
+	FinishTime metav1.MicroTime `json:"finishTime"`
+	// Error is a non-empty string if the build did not complete successfully.
+	Error string `json:"error,omitempty"`
+}
+
+// RuntimeStatus describes the current state of resource execution. Some values might not be used by every RuntimeType.
+type RuntimeStatus string
+
+const (
+	// RuntimeStatusPending indicates that the resource has not yet been started because it is waiting on another
+	// resource.
+	RuntimeStatusPending RuntimeStatus = "pending"
+	// RuntimeStatusRunning indicates that the resource is currently executing.
+	RuntimeStatusRunning RuntimeStatus = "running"
+	// RuntimeStatusSucceeded indicates that the resource ran to completion successfully.
+	//
+	// This is only used for a RuntimeType of "job"; "server" resources never exit during normal operation.
+	RuntimeStatusSucceeded RuntimeStatus = "succeeded"
+	// RuntimeStatusFailed indicates that the resource encountered an error during execution.
+	RuntimeStatusFailed RuntimeStatus = "failed"
+	// RuntimeStatusInactive indicates that the resource has been requested to not execute at this time.
+	RuntimeStatusInactive RuntimeStatus = "inactive"
```

Comment:
This covers things like `auto_init=False` today and would map well with being able to pause/stop resources from UI in the near-ish future.

That said, it's kind of funky and we could probably get rid of it by splitting resources into build/runtime sub-resources (see my other comment) because then disabled _runtime_ resources would just be "filtered out" and not exist in the list to begin with. (Of course, it's often useful to know "this thing exists but is disabled"... 🤔)

Personally, I don't hate it, but flagging in case anyone finds it unconscionable or has a more elegant solution up their sleeves

Reply:
maybe 'disabled' , since we use the words enable/disable resources elsewhere in the tiltfile api? but ya, it's a good call that we need this for resources that aren't expected to run in CI mode